### ISSUE TITLE
Case-insensitive globals changed from eslint-plugin-compat 6.2.0 to 7.0.0

### ISSUE THREAD

**Issue by joshkel** [amilajack/eslint-plugin-compat#684]
Case sensitivity has changed between eslint-plugin-compat 6.2.0 and 7.0.0 in ways that can result in false positives.

6.2.0 reported no issues for the following:

```
import * as serviceWorker from './serviceWorker.ts';
serviceWorker.register(false);
```

7.0.0 gives the following error:

> ServiceWorker is not supported in op_mini all, KaiOS 2.5, android 145

This isn't a reference to the `ServiceWorker` global - it's my own `serviceWorker` variable. Case sensitivity matters; eslint-plugin-compat shouldn't confuse the two. I believe that the fix in #681 is far too aggressive: the browser `crypto` global variable is an instance of the `Crypto` class, but there are plenty of other DOM API classes that do not follow this pattern (like `ServiceWorker`).

As another example:

```ts
   if (!navigator.permissions) {
      return;
    }
    // eslint-disable-next-line compat/compat
    navigator.permissions
      .query({ name: 'local-network-access' })
      .then((permissionStatus) => {
        permissionStatus.addEventListener('change', handleChange);
      })
```

I expect that I may have to suppress eslint-plugin-compat's warning when accessing `navigator.permissions`. However, starting with eslint-plugin-compat 7.0.0, I also have to suppress warnings about using my local `permissionStatus` variable, just because it happens to have the same name as a `PermissionStatus` interface that isn't globally recognized.

**Comment by vsalomaki** [comment]
You are right, the comparison is way too broad now. I'll make a proposal PR for the fix so that we could still maintain the Crypto & friends check

**Comment by vsalomaki** [comment]
@joshkel Can you see if that fix would work? I added tests per your example cases as well.

@amilajack Unfortunately this would need to be triaged quite soon, reverting the earlier commit from #681 or implementing fix

**Comment by joshkel** [comment]
@vsalomaki I just tested 7.0.1, and it's working now. Thanks for the quick fix!

**Comment by KuSh** [comment]
Is this also backported to 6.x branch? I'm having errors upgrading to 6.2.1, the following code errors on 6.2.1:

```ts
  decodeAudioData(audioData: ArrayBuffer): Promise<AudioBuffer> {
    return Promise.resolve({
      duration: audioData.byteLength / audioContextMock.sampleRate, // error  AudioData is not supported in Chrome 49  compat/compat
      length: audioData.byteLength, // error  AudioData is not supported in Chrome 49  compat/compat
      numberOfChannels: 1,
      sampleRate: audioContextMock.sampleRate,
      getChannelData() {
        return new Float32Array(audioData.byteLength); // error  AudioData is not supported in Chrome 49  compat/compat
      },
    } as unknown as AudioBuffer);
  },
```

Renaming the parameter silence the errors

**Comment by vsalomaki** [comment]
This fix has not been backported to 6.x branch, but it probably should be. What's your view @amilajack ?
